### PR TITLE
Add x86-64_full capability

### PR DESCRIPTION
**Pull request by DL6ER**
## Thank you for your contribution to the Pi-hole Community!

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**

 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 4. File a pull request for any change that requires changes to [our documentation](https://docs.pi-hole.net/) at our [documentation repo](https://github.com/pi-hole/docs) 

---

**What does this PR aim to accomplish?:**

Allow building a (nearly) all-options build of `dnsmasq`

**How does this PR accomplish the above?:**

Add remaining dependencies. We skip `i18n` and `ubus` as the former strongly depends on the user's system and makes nearly no difference code-wise (one macro is re-defined). `ubus` is only natively available on OpenWRT systems and not easily installable on any other distributions.

The `x86_64` container needs more time for building now as it has an additional stage to build such a full `dnsmasq` build *in addition* to the regular build. Both binaries are tested.

**Link documentation PRs if any are needed to support this PR:**

None

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [X] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by DL6ER** [comment]
The action on FTL's branch `special/CI_developmen is expected to fail the `_full` workflow path as this will only work once these containers are tagged and pushed.

**Review by yubiuser — COMMENTED** [review]
Shouldn't we move all the extra stuff that is needed only for the `all-in` below the `FROM builder AS all-in-compilation` line?

So it won't be included in the pushed "normal" build image?

__

I would also add a new "tester-all-in" step to be able to push an "all-in" image without the test remains.

**Comment by DL6ER** [comment]
Creating two containers (one large one, another one even larger) doesn't seem to make too much sense when we can just live with one (the even larger). It does not only save (total) space but also eases maintenance as we'd otherwise have two mostly identical Dockerfile where we need to update/change things.

The separation of the `builder` (https://github.com/pi-hole/docker-base-images/pull/57) should already cover this - or did I misinterpret something? We are pushing the `builder` and not the `all-in-compliation` target

File: ftl-build/x86_64/Dockerfile [hunk 1]
```diff
@@ -81,12 +92,65 @@ RUN curl -sSL https://ftl.pi-hole.net/libraries/nettle-${nettleversion}.tar.gz |
...
 # Install bats-core directly into the build image
 RUN git clone https://github.com/bats-core/bats-core.git \
     && cd bats-core
 
 ENV BATS=/bats-core/bin/bats
 
+FROM builder AS all-in-compilation
+
+# For FTL test compilation
+ARG CI_ARCH="x86_64_full"
```
Comment (yubiuser) [inline ftl-build/x86_64/Dockerfile:139]:
```suggestion
# Setting `CI_ARCH` to `x86_64_full` will trigger a compilation with extensive compiler options enabled within the makefile 
ARG CI_ARCH="x86_64_full"
```

**Review by yubiuser — COMMENTED** [review]
I think we should also test the "all-in" during image creation and not only during the builds in the FTL repo. Currently, our `ftl-build.yml` only build two targtes, `builder` and `tester`.
You can see it in the latest run, the "all-in" is not tested: https://github.com/pi-hole/docker-base-images/actions/runs/3443550395/jobs/5745202463

You would need to add a section like

```
      -
        name: Build all-in
        uses: docker/build-push-action@v3
        with:
          context: ftl-build/x86_64_full/.
          push: false
          target: all-in-compilation
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}

File: .github/workflows/ftl-build.yml [hunk 2]
```diff
@@ -71,6 +71,16 @@ jobs:
           # Caching disabled for now...
           # cache-from: type=gha,scope=${{ matrix.ARCH }}
           # cache-to: type=gha,scope=${{ matrix.ARCH }},mode=max
+      -
+        name: Build (all-in)
+        if: ${{ matrix.ARCH }} == 'x86_64'
```
Comment (yubiuser) [inline .github/workflows/ftl-build.yml:76]: Try

```suggestion
        if: matrix.ARCH == 'x86_64'
```

**Review by yubiuser — APPROVED** [review]

**Pull request merged**